player/lua/ytdl_hook.lua

```diff
@@ -57,6 +59,35 @@ local function edl_escape(url)
     return "%" .. string.len(url) .. "%" .. url
 end
 
+local function time_to_secs(time_string)
+    if not time_string then
+        return nil
+    end
+
+    local secs = 0
+    string.gsub(time_string, "[%d.]+", function (number)
+        secs = secs * 60 + tonumber(number)
+    end)
+
+    return secs
+end
+
+local function extract_chapters(data, video_length)
+    local ret = {}
+
+    for line in data:gmatch("[^\r\n]+") do
+        local time_string = string.match(line, "([%d:.]+:[%d.]+)")
```

Comment:
~~Basically it only cares about 01:23, 1:23:45, 1:2:34.56. Not sure if the milliseconds one works, but no point looking for patterns with only one series of digits.~~
Didn't notice the `:` in the character range.
The regex can be simpler, if this is only supposed to work with Youtube, according to:
![image](https://i.fsbn.eu/AtzY.png)